### PR TITLE
Add Round.apply function - Closes #3633

### LINKED ISSUE #3633: Add Round.apply function
### Expected behavior
We'll need to create a new Round module to support BFT consensus LIP.
### Actual behavior
The current rounds module is not compatible with the LIP
### Acceptance Criteria
- [x] If a new round is about to start, delegate list must be generated.
- [x] Increase producedBlocks on mem_accounts table
- [x] Apply votes to mem_accounts table
- [x] Update delegate ranks
- [ ] Continue publishing events in existing rounds module
- [x] If it's the end of the round, process rewards & missedBlocks information.
- [x] functions needs to be exported from `submodules/round/round.js` file for testing but only `apply` must be exported from `submodules/round/index.js`.

### PR DESCRIPTION
**Pull request by yatki**
### Review checklist

* The PR resolves #3633
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/dpos/apply.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		delegateReward,
+		isGettingRemainingFees,
+		round
+	) {
+		const exceptionRound = this.exceptions.rounds[round.toString()];
+		if (exceptionRound) {
+			// Apply rewards factor
+			delegateReward = delegateReward
+				.times(exceptionRound.rewards_factor)
+				.floor();
```
Comment (shuse2) [inline framework/src/modules/chain/dpos/apply.js]: Reward shouldn't be always int64? same as fee

File: framework/src/modules/chain/dpos/apply.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+				const account = await this.storage.entities.Account.get(
+					{ address },
+					{ extended: true },
+					tx
+				);
+
+				const data = distributeRewardsAndFees(account, earnings);
+
+				await this.storage.entities.Account.update({ address }, data, tx);
```
Comment (shuse2) [inline framework/src/modules/chain/dpos/apply.js]: can we gather all the information first and do the update all at once in one place?

File: framework/src/modules/chain/blocks/process.js [hunk 3]
```diff
@@ -60,34 +60,15 @@ class BlocksProcess {
 		}
 		if (typeof broadcast === 'function') {
 			broadcast(normalizedBlock);
+			await this.blocksVerify.checkExists(normalizedBlock);
```
Comment (shuse2) [inline framework/src/modules/chain/blocks/process.js]: why change this line?

File: framework/src/modules/chain/blocks/process.js [hunk 4]
```diff
@@ -188,7 +169,7 @@ class BlocksProcess {
 				continue;
 			}
 			// eslint-disable-next-line no-await-in-loop
-			lastBlock = await this.applyBlock(block, lastBlock);
+			lastBlock = await this.processBlock(block, lastBlock);
```
Comment (shuse2) [inline framework/src/modules/chain/blocks/process.js]: this should be apply. block should not be saved

File: framework/src/modules/chain/dpos/apply.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+					account,
+					data,
+				};
+			}
+		);
+
+		return Promise.all(updatedDelegates);
+	}
+
+	async updateVotes(updatedDelegates, tx) {
```
Comment (shuse2) [inline framework/src/modules/chain/dpos/apply.js]: updateVotes is done during the transaction execution, and we should keep it there

File: framework/src/modules/chain/dpos/common.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+const updateProducedBlocks = (address, undo = false) => {
```
Comment (shuse2) [inline framework/src/modules/chain/dpos/common.js]: It needs header, and `common` is not really the name it captures this functions.

File: framework/src/modules/chain/dpos/apply.js [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+			await this.updateMissedBlocks(nonForgedDelegateAddresses, tx);
+			const updatedDelegates = await this.distributeRewardsAndFees(
+				summarizedRound,
+				tx
+			);
+			await this.updateVotes(updatedDelegates, tx);
+		}
+	}
+
+	async updateProducedBlocks(block, tx) {
```
Comment (shuse2) [inline framework/src/modules/chain/dpos/apply.js]: if it's private function, it should be `_ updateProducedBlocks`
Same apply to belows

File: framework/src/modules/chain/dpos/apply.js (hunk 5 above)
Comment (yatki) [reply framework/src/modules/chain/dpos/apply.js]: As we discussed earlier, I'll rename the function to be more specific to rewards & fees and keep it here.

File: framework/src/modules/chain/blocks/process.js (hunk 4 above)
Comment (yatki) [reply framework/src/modules/chain/blocks/process.js]: I'll revert this change.

File: framework/src/modules/chain/blocks/process.js (hunk 3 above)
Comment (yatki) [reply framework/src/modules/chain/blocks/process.js]: I'll revert this change.

**Pull request closed without merging**